packages/flutter/lib/src/material/date_picker.dart

```diff
@@ -603,26 +601,34 @@ class _DatePickerDialogState extends State<_DatePickerDialog> {
   _DatePickerMode _mode = _DatePickerMode.day;
   GlobalKey _pickerKey = new GlobalKey();
 
+  void _vibrate() {
+    switch (Theme.of(context).platform) {
+      case TargetPlatform.android:
+      case TargetPlatform.fuchsia:
+        HapticFeedback.vibrate();
+        break;
+      case TargetPlatform.iOS:
+        break;
+    }
+  }
+
```

Comment:
If we find any other places where we put this switch, we should probably consider factoring it out into the theme or some such.